src/utils.ts

```diff
@@ -93,16 +79,23 @@ export async function getData({
   });
 
   if (!response.ok) {
-
-    const errorBody = await response.json();
-    const error = {
-      status: response.status,
-      message: errorBody.message || 'API request failed',
-    };
-
-    throw error;
+    throw (await makeError(response));
   }
 
   const result = await response.json();
   return result;
+}
+
+async function makeError(response: Response) {
+  let errorBody: any;
+  try {
+    errorBody = await response.json();
+  } catch (_) {
+    errorBody = null;
+  }
+
+  return {
+    status: response.status,
+    message: errorBody?.message || "API request failed",
```

Comment:
I think we should also add `success: false` to response object, to make response object parameters a bit more consistent.

```suggestion
    message: errorBody?.message || "API request failed",
    success: false,
```